### ISSUE TITLE
Unexpected Buffer behavior with broadcast assignment

### ISSUE THREAD

**Issue by rozsasarpi** [FluxML/Zygote.jl#854]
Is the below behavior expected or a bug?

```julia
Ab = Buffer([1.0], 2, 2)
Ab[:] = zeros(4)
Ab[[1, 2]] .+= [1.0, 2.0]
@show Ab

Buffer{Float64,Array{Float64,2}}([0.0 0.0; 0.0 0.0], false)
```

Without broadcasting it looks good:
```julia
Ab[[1, 2]] += [1.0, 2.0]
@show Ab

Buffer{Float64,Array{Float64,2}}([1.0 0.0; 2.0 0.0], false)
```

With a regular array it works as expected:
```julia
A = zeros(2, 2)
A[[1, 2]] .+= [1.0, 2.0]
@show A

2×2 Array{Float64,2}:
 1.0  0.0
 2.0  0.0
```

**Comment by ToucheSir** [comment]
Related issues: https://github.com/FluxML/Zygote.jl/issues/254, https://github.com/FluxML/Zygote.jl/pull/360

Looking at the lowering, `A[[1, 2]] .+= [1.0, 2.0]` calls `Base.maybeview`, which falls back to `getindex` on non-AbstractArrays. We could try `@forward`ing `maybeview` to the inner array, but I'm not sure what implications that would have for AD correctness.

**Comment by paulxshen** [comment]
bumping.. hny btw! is there plan to fix broadcast assignment? it still fails silently:
```
using Zygote: Buffer
a = Buffer([1], 2, 2)
a[:, 1] .= 8
a # a is unchanged :(
```

**Comment by ToucheSir** [comment]
I don't believe anything has changed. Frankly, I have a mind to close most of these Buffer-related issues as WONTFIX so that people are dissuaded from using it and don't expect any improvements to the interface. It has been a buggy mess from day one and anyone who needs mutation is better served by another AD (Enzyme, Mooncake, even ForwardDiff) these days.

**Comment by mcabbott** [comment]
Somehow Buffer is prominent on https://fluxml.ai/Zygote.jl/dev/limitations/ , maybe we should remove that (and encourage using map, comprehensions, custom rules?)
